slack_bolt/oauth/oauth_flow.py

```diff
@@ -216,29 +224,30 @@ def handle_callback(self, request: BoltRequest) -> BoltResponse:
             )
 
         # state parameter verification
-        state = request.query.get("state", [None])[0]
-        if not self.settings.state_utils.is_valid_browser(state, request.headers):
-            return self.failure_handler(
-                FailureArgs(
-                    request=request,
-                    reason="invalid_browser",
-                    suggested_status_code=400,
-                    settings=self.settings,
-                    default=self.default_callback_options,
+        if self.settings.state_validation_enabled is True:
+            state = request.query.get("state", [None])[0]
+            if not self.settings.state_utils.is_valid_browser(state, request.headers):
+                return self.failure_handler(
+                    FailureArgs(
+                        request=request,
+                        reason="invalid_browser",
+                        suggested_status_code=400,
+                        settings=self.settings,
+                        default=self.default_callback_options,
+                    )
                 )
-            )
 
-        valid_state_consumed = self.settings.state_store.consume(state)
-        if not valid_state_consumed:
-            return self.failure_handler(
-                FailureArgs(
-                    request=request,
-                    reason="invalid_state",
-                    suggested_status_code=401,
-                    settings=self.settings,
-                    default=self.default_callback_options,
+            valid_state_consumed = self.settings.state_store.consume(state)
+            if not valid_state_consumed:
+                return self.failure_handler(
```

Comment:
Codecov says that this line is not covered by tests, but I see a variety of `test_handle_callback_invalid_state` tests. However, these tests do not assert that the response code is 401, nor that the reason is `invalid_state`. Are [these particular `invalid_state` tests](https://github.com/slackapi/bolt-python/blob/main/tests/slack_bolt/oauth/test_oauth_flow.py#L144) exercising a different code path, then? It may be useful to add those tests to this pull request.

Reply:
Thanks for pointing this out. I see the current test covers L231, not this pattern. I will add more tests before merging this PR